modules/apps/27-interchain-accounts/host/keeper/events.go

```diff
@@ -18,3 +18,14 @@ func EmitWriteErrorAcknowledgementEvent(ctx sdk.Context, packet exported.PacketI
 		),
 	)
 }
+
+// EmitSuccessfulAcknowledgementEvent emits an event signalling a successful acknowledgement
+func EmitSuccessfulAcknowledgementEvent(ctx sdk.Context, packet exported.PacketI) {
+	ctx.EventManager().EmitEvent(
+		sdk.NewEvent(
+			icatypes.EventTypePacket,
+			sdk.NewAttribute(sdk.AttributeKeyModule, icatypes.ModuleName),
+			sdk.NewAttribute(icatypes.AttributeKeyHostChannelID, packet.GetDestChannel()),
```

Comment:
I would like input on whether or not there are additional fields/attributes that should be included in this event. Please let me know if I've missed anything!

Reply:
This looks fine, maybe we could add a `AttributeKeyAckSuccess` [like for transfer](https://github.com/cosmos/ibc-go/blob/25767f6bdb5bab2c2a116b41d92d753c93e18121/modules/apps/transfer/ibc_module.go#L202).

But I am also now wondering if we could merge both event functions into one and have something like this:

```
ctx.EventManager().EmitEvent(
	sdk.NewEvent(
		icatypes.EventTypePacket,
		sdk.NewAttribute(sdk.AttributeKeyModule, icatypes.ModuleName),
		sdk.NewAttribute(icatypes.AttributeKeyAckError, err.Error()),
		sdk.NewAttribute(icatypes.AttributeKeyHostChannelID, packet.GetDestChannel()),
                sdk.NewAttribute(icatypes.AttributeKeyAckSuccess, fmt.Sprintf("%t", ack.Success())),
	),
)
```

with the `icatypes.AttributeKeyAckError` empty if it was successful.

Reply:
I was actually thinking something very similar with regards to merging the functions.